pyccel/codegen/printing/ccode.py

```diff
@@ -587,6 +603,67 @@ def function_signature(self, expr):
         else:
             return '{0}{1}({2})'.format(ret_type, name, arg_code)
 
+    def _print_IndexedElement(self, expr):
+        if isinstance(expr.base, IndexedVariable):
+            base = expr.base.internal_variable
+        else:
+            base = expr.base
+        inds = list(expr.indices)
+        inds = inds[::-1]
+        base_shape = base.shape
+        allow_negative_indexes = (isinstance(base, IndexedVariable) and \
+                base.internal_variable.allows_negative_indexes)
+        base = self._print(expr.base.label)
+        for i, ind in enumerate(inds):
+            if isinstance(ind, PyccelUnarySub) and isinstance(ind.args[0], LiteralInteger):
+                inds[i] = PyccelMinus(base_shape[i], ind.args[0])
+            else:
+                if isinstance(ind, Slice):
+                    #setting the slice start and end to their correct value if none is provided
+                    start = ind.start
+                    end = ind.end
+                    if ind.start is None:
+                        start = 0
+                    if ind.end is None:
+                        end = expr.base.shape[1]
```

Comment:
Beware of using the shape of an `IndexedVariable`. It seems to be defined as the shape of the underlying `IndexedBase`. `IndexedBase` is a sympy object to there may be data loss when saving this value. You would be better off using the base variable you defined above which contains the `internal_variable` if the object is an `IndexedVariable`

Reply:
fixed